### DISCUSSION TITLE
Response headers / cookies

### DISCUSSION THREAD

**Discussion by lpuzzolokodear** [Textalk/websocket-php#176] [Q&A]
Hi!
How do you get the response headers?
On some services, cookies are set on the response (ie, session cookies).

Thanks in advance!
Lisandro.

**Comment by sirn-se** [comment]
Do you have an example of such service?

Adding custom headers to the handshake seems a bit odd, and nothing I have encountered before. So it would be good with a real world example where HTTP headers are used in this way.

/Sören

**Reply by lpuzzolokodear** [reply]
Hi!
For example some Gradio WSS apps return a session cookie that you need to use in further requests in order to retrieve the generated data. 
If you generate images via WSS, the service returns the URL of the generated images, but you can only get the image content using requests on the same http session. So you need to get the cookie from the WSS response, and use that cookie on the following HTTP requests.

![image](https://user-images.githubusercontent.com/77172361/220374239-9dbbfc5d-a067-4d1c-b084-64342c3f6ad1.png)

I did manage to obtain the session cookie through an alternative mechanism, so if it's an odd task, please don't worry about having to retrieve response headers as shown in the attached image.

Thank you very much for your response!

**Comment by sirn-se** [comment]
It is a bit strange, as the session cookie have no purpose within the websocket connection but apparently is supposed to be used on HTTP requests run elsewhere in the code.

But the ability to retrieve headers is a valid request in itself. As of now we only support setting headers, but I will add this feature to my todo list.

Please also note that I maintain this repo as [phrity/websocket](https://github.com/sirn-se/websocket-php), any updates will end up there instead.
